src/MetricsTimer.h

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+#pragma once
+#include "Forwarder.h"
+#include "Kafka.h"
+#include "MainOpt.h"
+#include "logger.h"
+#include <asio.hpp>
+#include <atomic>
+#include <chrono>
+#include <functional>
+#include <mutex>
+#include <thread>
+#include <vector>
+
+namespace Forwarder {
+
+using CallbackFunction = std::function<void()>;
+
+/// Multiple callbacks can be registered with the Timer, which will repeatedly
+/// execute them at a set interval
+class MetricsTimer {
+public:
+  explicit MetricsTimer(std::chrono::milliseconds Interval,
+                        MainOpt &ApplicationMainOptions,
+                        std::atomic<std::chrono::milliseconds>
+                            &MainLoopIterationExecutionDuration,
```

Comment:
I do not think we need to have `MainLoopIterationExecutionDuration`. Removing it reduces the number of shared variables and it feels like it is something the PSI guys used for debugging some specific problems anyway.

Reply:
I assume the idea was that it would indicate whether the forwarder is overloaded with PVs and can no longer keep up. However, I'm not against removing it given that we are not currently monitoring it and I don't know if it is actually a reliable indicator of that anyway.